x-pack/test/security_solution_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,197 @@
+  const pageObjects = getPageObjects(['common', 'timePicker', 'hosts', 'settings']);
+  const testSubjects = getService('testSubjects');
+  const esArchiver = getService('esArchiver');
+
+  describe('Endpoint Event Resolver', function () {
+    this.tags(['ciGroup7']);
+
+    before(async () => {
+      await esArchiver.load('endpoint/resolver_tree/api_feature', { useCreate: true });
+      await pageObjects.hosts.navigateToSecurityHostsPage();
+      const fromTime = 'Jan 1, 2018 @ 00:00:00.000';
+      const toTime = 'now';
+      await pageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await (await testSubjects.find('draggable-content-host.name')).click();
+      await testSubjects.existOrFail('header-page-title');
+      await (await testSubjects.find('navigation-events')).click();
+      await testSubjects.existOrFail('events-viewer-panel');
+      await (await testSubjects.findAll('investigate-in-resolver-button'))[0].click();
+      await testSubjects.exists('investigate-in-resolver-button', { timeout: 4000 });
+    });
+
+    it('check that Resolver and Data table is loaded', async () => {
+      await testSubjects.existOrFail('resolver:graph');
+      await testSubjects.existOrFail('tableHeaderCell_name_0');
+      await testSubjects.existOrFail('tableHeaderCell_timestamp_1');
+    });
+
+    it('compare resolver Nodes Table data and Data length', async () => {
+      const nodeData: string[] = [];
+      const TableData: string[] = [];
+
+      const Table = await testSubjects.findAll('resolver:node-list:item');
+      for (const value of Table) {
+        const text = await value._webElement.getText();
+        TableData.push(text.split('\n')[0]);
+      }
+
+      await testSubjects.click('resolver:graph-controls:zoom-out');
+      const Nodes = await testSubjects.findAll('resolver:node:primary-button');
+      for (const value of Nodes) {
+        nodeData.push(await value._webElement.getText());
+      }
+      for (let i = 0; i < nodeData.length; i++) {
+        expect(TableData[i]).to.eql(nodeData[i]);
+      }
+      expect(nodeData.length).to.eql(TableData.length);
+      await testSubjects.click('resolver:graph-controls:zoom-in');
+    });
+
+    it('resolver Nodes navigation Up', async () => {
+      const OriginalNodeDataStyle = await pageObjects.hosts.parseStyles();
+      await (await testSubjects.find('resolver:graph-controls:north-button')).click();
+
+      const NewNodeDataStyle = await pageObjects.hosts.parseStyles();
+      for (let i = 0; i < OriginalNodeDataStyle.length; i++) {
+        expect(parseFloat(OriginalNodeDataStyle[i].top)).to.lessThan(
+          parseFloat(NewNodeDataStyle[i].top)
+        );
+        expect(parseFloat(OriginalNodeDataStyle[i].left)).to.equal(
+          parseFloat(NewNodeDataStyle[i].left)
+        );
+      }
+      await (await testSubjects.find('resolver:graph-controls:center-button')).click();
```

Comment:
Is there something special with resolver:graph-controls:center-button that you need to find it first before clicking on it or can you click directly?

Reply:
No
this is the way we should click on the buttons

Reply:
I notice you have 2 styles:
Find and click
await (await testSubjects.find('resolver:graph-controls:center-button')).click();

and

Click
await testSubjects.click('resolver:graph-controls:south-button');

Reply:
This is what Dima told to use

Reply:
Is there a reason to use both styles or can you stick with one?  Is one for a particular scenario and the other for another scenario?